src/glados/core/tts_kokoro.py

```
@@ -0,0 +1,96 @@
+        symbols = [_pad, *_punctuation, *_letters, *_letters_ipa]
+        dicts = {}
+        for i in range(len(symbols)):
+            dicts[symbols[i]] = i
+        return dicts
+
+    def _phonemes_to_ids(self, phonemes: str) -> list[int]:
+        if len(phonemes) > MAX_PHONEME_LENGTH:
+            raise ValueError(f"text is too long, must be less than {MAX_PHONEME_LENGTH} phonemes")
+        return [i for i in map(self.vocab.get, phonemes) if i is not None]
+
+    def _synthesize_ids_to_audio(self, ids: list[int], voice: str = None) -> NDArray[np.float32]:
```

Ruff (0.8.2)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)